content/en/docs/concepts/services-networking/topology-aware-hints.md

```diff
@@ -0,0 +1,127 @@
+the zone it originated from. This concept is commonly referred to as "Topology
+Aware Routing". The "Hints" here refer to a key implementation detail. The
+EndpointSlice controller adds hints to EndpointSlices to allocate each endpoint
+to a zone. Those hints are then consumed by components like kube-proxy as they
+configure how requests are routed.
+
+## Using Topology Aware Hints
+
+This feature can be enabled for individual Services by setting the
+`service.kubernetes.io/topology-aware-hints` annotation to `auto`. This tells
+the EndpointSlice controller to set topology hints if it is deemed safe.
+Importantly, this does not guarantee that hints will always be set.
+
+## How it Works
+
+The functionality enabling this feature is split into two components: The
+EndpointSlice controller and Kube-Proxy. This provides a high level overview of
+how each component implements this feature.
+
+### EndpointSlice controller
+
+The EndpointSlice controller is responsible for setting hints on EndpointSlices
+when this feature is enabled. The controller allocates a proportional amount of
+endpoints to each zone. This proportion is based on the number of CPU cores that
+have been allocated to nodes running in that zone. For example, if one zone had
+2 CPU cores and another zone only had 1 CPU core, the controller would allocated
+twice as many endpoints to the zone with 2 CPU cores.
+
+### Kube-Proxy
+
+Kube-Proxy filters the endpoints it routes to based on the hints set by the
+EndpointSlice controller. Essentially, the controller allocates endpoints to
+specific zones, and kube-proxy is responsible for ensuring that it only routes
+to endpoints with hints for the zone it is in.
+
+## Safeguards
+
+Both the EndpointSlice controller and Kube-Proxy implementations have a number
+of safeguards built in that will result in falling back to the original
+approach. This results in traffic being routed to all endpoints, instead of just
+a topology aware subset. Here's what could trigger that:
+
+1. **Insufficient number of endpoints:** If there are less endpoints than zones
+   in a cluster, the controller will not assign any hints. 
+
+2. **Impossible to achieve balanced allocation:** In some cases, it will be
+   impossible to achieve a balanced allocation of endpoints among zones. For
+   example, if zone-a is twice as large as zone-b, but there are only 2
+   endpoints, an endpoint allocated to zone-a may receive twice as much traffic
+   as zone-b. The controller wil not assign hints if it can't get this "expected
+   overload" value below 20% for each zone. Importantly this is not based on
+   real-time feedback. It is still possible for individual endpoints to become
+   overloaded.
+
+3. **One or more Nodes has insufficient information:** If a node does not have a
+   `topology.kubernetes.io/zone` label or is not reporting a value for
+   allocatable CPU, this functionality will be disabled for all Services in a
+   cluster.
```

Comment:
nit: using the active voice, we can explain _what_ does not act:
```suggestion
   allocatable CPU, the control plane does not set any topology-aware endpoint hints.
```

Reply:
Updated this, but not sure if it's clear enough yet. I'm not sure everyone will be able to connect "does not set any topology-aware hints" to "this feature will not do anything at all".

Reply:
could add
> and so the kube-proxy selects endpoints from anywhere in your cluster, regardless of the zone.

Reply:
Updated, thanks!